apps/server/src/modules/provisioning/strategy/tsp/tsp.strategy.spec.ts

```diff
@@ -55,8 +62,121 @@ describe('TspProvisioningStrategy', () => {
 
 	describe('getData', () => {
 		describe('When called', () => {
-			it('should throw', () => {
-				expect(() => sut.getData({} as OauthDataStrategyInputDto)).toThrow();
+			const setup = () => {
+				const input: OauthDataStrategyInputDto = new OauthDataStrategyInputDto({
+					system: new ProvisioningSystemDto({
+						systemId: 'externalSchoolId',
+						provisioningStrategy: SystemProvisioningStrategy.TSP,
+					}),
+					idToken: 'tspIdToken',
+					accessToken: 'tspAccessToken',
+				});
+
+				jest.spyOn(jwt, 'decode').mockImplementation(() => {
+					return {
+						sub: 'externalUserId',
+						sid: 'externalSchoolId',
+						ptscListRolle: 'teacher',
+						personVorname: 'firstName',
+						personNachname: 'lastName',
+						ptscSchuleNummer: 'externalSchoolId',
+						ptscListKlasseId: ['externalClassId1', 'externalClassId2'],
+					};
+				});
+
+				const user: ExternalUserDto = new ExternalUserDto({
+					externalId: 'externalUserId',
+					roles: [RoleName.TEACHER],
+					firstName: 'firstName',
+					lastName: 'lastName',
+				});
+
+				const school: ExternalSchoolDto = new ExternalSchoolDto({
+					externalId: 'externalSchoolId',
+				});
+
+				const externalClass1 = new ExternalClassDto({ externalId: 'externalClassId1' });
+				const externalClass2 = new ExternalClassDto({ externalId: 'externalClassId2' });
+				const externalClasses = [externalClass1, externalClass2];
+
+				return { input, user, school, externalClasses };
+			};
+
+			it('should return mapped oauthDataDto if input is valid', async () => {
+				const { input, user, school, externalClasses } = setup();
+				const result = await sut.getData(input);
+
+				expect(result).toEqual({
```

Comment:
Better: `expect(result).toEqual<OauthDataDto>(...)`

Reply:
it can be removed also i just added it at first to control the values that return, done